src/pages/patientView/mutation/column/FACETSColumnFormatter.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {If, Else, Then } from 'react-if';
+import DefaultTooltip from "public-lib/components/defaultTooltip/DefaultTooltip";
+import 'rc-tooltip/assets/bootstrap_white.css';
+import {Mutation} from "shared/api/generated/CBioPortalAPI";
+import SampleManager from "../../sampleManager";
+import {isUncalled} from 'shared/lib/MutationUtils';
+
+export default class FACETSColumnFormatter {
+    static barWidth = 6;
+    static barSpacing = 3;
+    static maxBarHeight = 12;
+    static indexToBarLeft = (n:number) => n*(FACETSColumnFormatter.barWidth + FACETSColumnFormatter.barSpacing);
+
+    public static getComponentForSampleArgs<T extends {alleleSpecificCopyNumber:{ccfMCopies:number}}>(mutation:T) {
+        let opacity: number = 1;
+        let extraTooltipText: string = '';
+        if (mutation.alleleSpecificCopyNumber !== undefined && mutation.alleleSpecificCopyNumber.ccfMCopies !== undefined) {
+            const ccfMCopiesValue = mutation.alleleSpecificCopyNumber.ccfMCopies;
+            if (ccfMCopiesValue !== 1) {
+                opacity = .5;
+            }
+        }
+        return {
+            opacity,
+            extraTooltipText
+         };
+    }
+
+    public static convertMutationToSampleElement<T extends {sampleId:string, alleleSpecificCopyNumber:{ccfMCopies:number}}>(mutation:T, color:string, barX:number, sampleComponent:any) {
+        if (mutation.alleleSpecificCopyNumber !== undefined && mutation.alleleSpecificCopyNumber.ccfMCopies !== undefined) {
+            const ccfMCopies = mutation.alleleSpecificCopyNumber.ccfMCopies;
+            const barHeight = (isNaN(ccfMCopies) ? 0 : ccfMCopies)*FACETSColumnFormatter.maxBarHeight;
+            const barY = FACETSColumnFormatter.maxBarHeight - barHeight;
+
+            const bar = (<rect x={barX} y={barY} width={FACETSColumnFormatter.barWidth} height={barHeight} fill={color}/>);
+            const text = (<span>
+                    <strong>{ccfMCopies.toFixed(2)}</strong>
+                </span>);
+            return {
+                sampleId:mutation.sampleId, bar, component:sampleComponent, text, ccfMCopies
+            };
+        }
+    }
+
+    public static renderFunction(mutations:Mutation[], sampleManager:SampleManager|null) {
+        if (!sampleManager) {
+            return (<span></span>);
+        }
+
```

Comment:
can you briefly summarize what's going on here since there seems to be a lot going on?